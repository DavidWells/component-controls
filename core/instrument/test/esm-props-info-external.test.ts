import path from 'path';
import { loadStoriesTests, extractComponentProps } from './loadTestFiles';

describe('external-library', () => {
  loadStoriesTests(
    {
      stories: { storeSourceFile: false },
      propsLoaders: [
        { name: '@component-controls/react-docgen-info', test: /\.(js|jsx)$/ },
        {
          name: '@component-controls/react-docgen-typescript-info',
          test: /\.(ts|tsx)$/,
        },
        ,
      ],
      components: {
        storeSourceFile: false,
        resolveFile: (componentName, filePath) => {
          if (filePath.includes('theme-ui/dist')) {
            const resolved = path.resolve(
              path.dirname(filePath),
              `../../@theme-ui/components/index.d.ts`,
            );
            return resolved;
          }
          return filePath;
        },
      },
    },
    ['esm', 'props-info-external'],
  );
  extractComponentProps(
    'Button',
    path.resolve(
      __dirname,
      '../../../node_modules/@theme-ui/components/index.d.ts',
    ),
  );
});
