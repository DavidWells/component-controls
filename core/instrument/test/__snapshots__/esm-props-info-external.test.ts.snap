// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`external-library theme-ui.jsx 1`] = `
Object {
  "components": Object {
    "00be19a017a200d882dd02db4137b6bf": Object {
      "from": "theme-ui",
      "importedName": "Button",
      "imports": undefined,
      "name": "Button",
      "request": undefined,
    },
  },
  "doc": Object {
    "component": "Button",
    "components": Object {},
    "componentsLookup": Object {
      "Button": "00be19a017a200d882dd02db4137b6bf",
    },
    "date": 2020-08-05T12:08:58.984Z,
    "dateModified": 2020-08-05T12:13:10.991Z,
    "package": "0de7221ca26b66ff42f614448352017c",
    "title": "Storybook/Blocks/Button",
  },
  "packages": Object {
    "0de7221ca26b66ff42f614448352017c": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.16.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.10.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "0de7221ca26b66ff42f614448352017c",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/props-info-external/theme-ui.jsx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.16.0",
    },
  },
  "stories": Object {},
  "transformed": "import { Button } from 'theme-ui';

export default {
  title: 'Storybook/Blocks/Button',
  component: Button,
};
",
}
`;
